### PR TITLE
Feature/ELG-16/API for refrigerators with a direct sale funtion

### PR DESCRIPTION
**Pull request by FernandoRamirez-De-Aguilar**
- API endpoints for energy label and internet of
  1. Refrigerated vending machines
  2. Ice cream freezers
  3. Beverage coolers
  4. Supermarket refrigerator/freezer cabinets or gelato-scooping cabinets

- Added setters/getters to labelColour in RescaledInternetLabelApiForm
- Added generateInternetLabel method with a new signature that takes a RescaledInternetLabelApiForm to handle the label colour logic

### CODE REVIEW

File: src/main/java/uk/gov/beis/els/api/categories/refrigeratordirectsales/BeverageCoolersInternetLabelApiForm.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package uk.gov.beis.els.api.categories.refrigeratordirectsales;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import javax.validation.constraints.NotBlank;
+import uk.gov.beis.els.api.common.ApiValuesFromLegislationCategory;
+import uk.gov.beis.els.api.common.RescaledInternetLabelApiForm;
+import uk.gov.beis.els.categories.refrigeratorsdirectsales.service.RefrigeratorsDirectSalesService;
+
+@Schema(name = "Beverage coolers internet label")
```
Comment (FernandoRamirez-De-Aguilar) [inline src/main/java/uk/gov/beis/els/api/categories/refrigeratordirectsales/BeverageCoolersInternetLabelApiForm.java:9]: all internet label API forms are the same, but I couldn't come with a way of making the schema name different for each category.
Happy to reworked if there is a solution to that

File: src/main/java/uk/gov/beis/els/api/categories/refrigeratordirectsales/BeverageCoolersInternetLabelApiForm.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import io.swagger.v3.oas.annotations.media.Schema;
+import javax.validation.constraints.NotBlank;
+import uk.gov.beis.els.api.common.ApiValuesFromLegislationCategory;
+import uk.gov.beis.els.api.common.RescaledInternetLabelApiForm;
+import uk.gov.beis.els.categories.refrigeratorsdirectsales.service.RefrigeratorsDirectSalesService;
+
+@Schema(name = "Beverage coolers arrow image")
+public class BeverageCoolersInternetLabelApiForm extends RescaledInternetLabelApiForm {
+
+  @Schema(description = "Energy efficiency class indicator")
```
Comment (dannybetts) [inline src/main/java/uk/gov/beis/els/api/categories/refrigeratordirectsales/BeverageCoolersInternetLabelApiForm.java:12]: Should these say something more along the lines of "Energy efficiency class for beverage coolers"? And so on with the other ones.

File: src/main/java/uk/gov/beis/els/api/categories/refrigeratordirectsales/RefrigeratorsDirectSalesApiController.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+public class RefrigeratorsDirectSalesApiController {
+
+  private final RefrigeratorsDirectSalesService refrigeratorsDirectSalesService;
+  private final InternetLabelService internetLabelService;
+  private final DocumentRendererService documentRendererService;
+
+  @Autowired
+  public RefrigeratorsDirectSalesApiController(
+      RefrigeratorsDirectSalesService refrigeratorsDirectSalesService,
+      InternetLabelService internetLabelService, DocumentRendererService documentRendererService) {
```
Comment (dannybetts) [inline src/main/java/uk/gov/beis/els/api/categories/refrigeratordirectsales/RefrigeratorsDirectSalesApiController.java:32]: Nit: formatting on constuctor

**Review by dannybetts — CHANGES_REQUESTED** [review]
Update wording on energy efficiency doco

File: src/main/java/uk/gov/beis/els/api/categories/refrigeratordirectsales/BeverageCoolersInternetLabelApiForm.java (hunk 2 above)
Comment (FernandoRamirez-De-Aguilar) [reply src/main/java/uk/gov/beis/els/api/categories/refrigeratordirectsales/BeverageCoolersInternetLabelApiForm.java]: addressed

File: src/main/java/uk/gov/beis/els/api/categories/refrigeratordirectsales/RefrigeratorsDirectSalesApiController.java (hunk 3 above)
Comment (FernandoRamirez-De-Aguilar) [reply src/main/java/uk/gov/beis/els/api/categories/refrigeratordirectsales/RefrigeratorsDirectSalesApiController.java]: thanks, addressed

File: src/main/java/uk/gov/beis/els/api/categories/refrigeratordirectsales/BeverageCoolersInternetLabelApiForm.java (hunk 1 above)
Comment (jamesbarnett91) [reply src/main/java/uk/gov/beis/els/api/categories/refrigeratordirectsales/BeverageCoolersInternetLabelApiForm.java]: We could just have 1 internet label form for all these categories, and call the schema something like `Arrow image for refrigerators with a direct sales function`. The single field could just be called `Energy efficiency class indicator` to keep it generic, which is what the current HTML forms do for this category.

Not sure if @matteason has an opinion on which is better?

**Review by jamesbarnett91 — CHANGES_REQUESTED** [review]
Few minor comments, bulk of it looks good though.

File: src/main/java/uk/gov/beis/els/api/common/RescaledInternetLabelApiForm.java [hunk 4]
```diff
@@ -6,7 +6,15 @@
 
 public class RescaledInternetLabelApiForm extends BaseInternetLabelApiForm {
 
-  @Schema(name = "Should the arrow be in colour or black and white?", description = "Use a colour arrow if you can. You can use a black and white arrow if your material is being printed in black and white. You shouldn't use black and white arrows on the internet.")
+  @Schema(description = "Use a colour arrow if you can. You can use a black and white arrow if your material is being printed in black and white. You shouldn't use black and white arrows on the internet.")
```
Comment (jamesbarnett91) [inline src/main/java/uk/gov/beis/els/api/common/RescaledInternetLabelApiForm.java:9]: I think its worth keeping the `name` field here, but we can update the text to me more clear, e.g. `name=The colour of the arrow image`

**Review by dannybetts — APPROVED** [review]

**Comment by dannybetts** [comment]
@FernandoRamirez-De-Aguilar Approved pending the decision on internet label form classes

File: src/main/java/uk/gov/beis/els/api/categories/refrigeratordirectsales/BeverageCoolersInternetLabelApiForm.java (hunk 1 above)
Comment (FernandoRamirez-De-Aguilar) [reply src/main/java/uk/gov/beis/els/api/categories/refrigeratordirectsales/BeverageCoolersInternetLabelApiForm.java]: I agree with the redundancy, will wait on @matteason opinion to make changes

File: src/main/java/uk/gov/beis/els/api/common/RescaledInternetLabelApiForm.java (hunk 4 above)
Comment (FernandoRamirez-De-Aguilar) [reply src/main/java/uk/gov/beis/els/api/common/RescaledInternetLabelApiForm.java:9]: if we define a name, the doco thinks that is the name of the paramenter. I removed it to display the param name as it should be defined
image
^ defining a name

Comment (jamesbarnett91) [reply src/main/java/uk/gov/beis/els/api/common/RescaledInternetLabelApiForm.java:9]: Ah sorry my bad, I see what you mean. Can we add to the first part of the description then please? E..g "The colour of the arrow image. Use a colour arrow if you can. ...etc"

**Review by jamesbarnett91 — APPROVED** [review]
Thanks for the changes. Looks good

**Pull request merged**